functional_test.sh

```diff
@@ -35,23 +39,36 @@ docker run \
     -classpath web/target/graphhopper-web-1.0-SNAPSHOT.jar com.graphhopper.http.GraphHopperApplication gtfs_links default_gh_config.yaml
 
 # Run server in background
-docker run --rm  --name functional_test_server -p 50051:50051 -p 8998:8998 -v "$TMPDIR:/graphhopper/transit_data/" \
+docker run --rm --name functional_test_server -p 50051:50051 -p 8998:8998 -v "$TMPDIR:/graphhopper/transit_data/" \
     "$DOCKER_IMAGE_TAG" &
 
 echo "Waiting for graphhopper server to start up"
-sleep 30
+sleep 30 # should we implement a trie mechanism with time threshold instead of sleep?
+
+# greb the server ip:
+SERVER= ` docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' functional_test_server }}`
 
 # Make a request for a couple of points in the minikc region
-grpcurl -d @ -plaintext localhost:50051 router.Router/RouteStreetMode > "$TMPDIR"/street_response.json <<EOM
+grpcurl  -d @ -plaintext $SERVER:50051 router.Router/RouteStreetMode > "$TMPDIR"/street_response.json <<EOM
 {
 "points":[{"lat":38.96637569955874,"lon":-94.70833304570988},{"lat":38.959204519370815,"lon":-94.69174071738964}],
 "profile": "car"
 }
 EOM
 
+#test if the client command exit status == 0
```

Comment:
Maybe not a huge deal, but have you confirmed that this works? I used the `set` command at the top of the script and thought that I configured it to exit immediately if any command failed

Reply:
yup